### PR TITLE
Minor fixes to prometheus snapshot

### PR DESCRIPTION
**Pull request by RobertKrawitz**
Also improve logging messages

### CODE REVIEW

File: benchmark_runner/benchmark_operator/benchmark_operator_workloads.py [hunk 1]
```diff
@@ -821,21 +822,25 @@ def run_workload(self, workload: str):
...
             artifacts_path = self.__environment_variables_dict.get('run_artifacts_path', '')
             if not os.path.isdir(artifacts_path):
                 os.mkdir(artifacts_path)
-            snapshot = PrometheusSnapshot(oc=self.__oc, artifacts_path=artifacts_path, verbose=True)
-            snapshot.prepare_for_snapshot()
+            try:
+                snapshot = PrometheusSnapshot(oc=self.__oc, artifacts_path=artifacts_path, verbose=True)
+                snapshot.prepare_for_snapshot()
+            except PrometheusSnapshotError as err:
+                raise PrometheusSnapshotFailed(err)
```
Comment (ebattat) [inline benchmark_runner/benchmark_operator/benchmark_operator_workloads.py:833]: Add also [Except Exception](https://github.com/redhat-performance/benchmark-runner/blob/8c7f55f8e2504e6cab1348659430497668bef238/benchmark_runner/benchmark_operator/benchmark_operator_workloads.py#L460)

File: benchmark_runner/benchmark_operator/benchmark_operator_workloads.py [hunk 2]
```diff
@@ -821,21 +822,25 @@ def run_workload(self, workload: str):
...
         self.run_workload_func(workload_full_name=workload)
 
         # Retrieve the Prometheus snapshot
-        if extract_prometheus_snapshot.lower() == 'true':
-            snapshot.retrieve_snapshot()
+        if self.__enable_prometheus_snapshot:
+            try:
+                snapshot.retrieve_snapshot()
+            except PrometheusSnapshotError as err:
+                raise PrometheusSnapshotFailed(err)
```
Comment (ebattat) [inline benchmark_runner/benchmark_operator/benchmark_operator_workloads.py:843]: Add also Except Exception

File: benchmark_runner/benchmark_operator/benchmark_operator_workloads.py [hunk 3]
```diff
@@ -821,21 +822,29 @@ def run_workload(self, workload: str):
...
-            snapshot.prepare_for_snapshot()
+            try:
+                if not os.path.isdir(artifacts_path):
+                    os.mkdir(artifacts_path)
+                snapshot = PrometheusSnapshot(oc=self.__oc, artifacts_path=artifacts_path, verbose=True)
+                snapshot.prepare_for_snapshot()
+            except PrometheusSnapshotError as err:
+                raise PrometheusSnapshotFailed(err)
+            except Exception as err:
+                raise PrometheusSnapshotFailed(err)
```
Comment (ebattat) [inline benchmark_runner/benchmark_operator/benchmark_operator_workloads.py:835]: I think it should be just err, we dont know if that relate to [Prometheus](https://github.com/redhat-performance/benchmark-runner/blob/8c7f55f8e2504e6cab1348659430497668bef238/benchmark_runner/benchmark_operator/benchmark_operator_workloads.py#L466)

File: benchmark_runner/benchmark_operator/benchmark_operator_workloads.py [hunk 4]
```diff
@@ -821,21 +822,29 @@ def run_workload(self, workload: str):
...
         # Retrieve the Prometheus snapshot
-        if extract_prometheus_snapshot.lower() == 'true':
-            snapshot.retrieve_snapshot()
+        if self.__enable_prometheus_snapshot:
+            try:
+                snapshot.retrieve_snapshot()
+            except PrometheusSnapshotError as err:
+                raise PrometheusSnapshotFailed(err)
+            except Exception as err:
+                raise PrometheusSnapshotFailed(err)
```
Comment (ebattat) [inline benchmark_runner/benchmark_operator/benchmark_operator_workloads.py:847]: I think it should be just err, we dont know if that relate to Prometheus

**Comment by RobertKrawitz** [comment]
All changes made, and e2e passes with snapshot both enabled and disabled, and verified both conditions.

**Review by ebattat — APPROVED** [review]
/LGTM
/Approved

**Pull request merged**